src/main/java/com/powsybl/openloadflow/sa/AcSecurityAnalysis.java

```diff
@@ -122,8 +144,40 @@ public static void distributedMismatch(LfNetwork network, double mismatch, LoadF
         }
     }
 
+    public static void getAllSwitchesToOperate(Network network, List<Action> actions, Set<Switch> allSwitchesToClose, Set<Switch> allSwitchesToOpen) {
+        actions.stream().filter(action -> action.getType().equals(SwitchAction.NAME))
+                .map(action -> ((SwitchAction) action).getSwitchId())
+                .forEach(id -> {
+                    Switch sw = network.getSwitch(id);
+                    if (sw.isOpen()) {
+                        allSwitchesToClose.add(sw);
```

Comment:
How do we known that the switch will be closed if not looking at action state?

Reply:
Right, thanks!